File: 2022-2023/sourcecode/lab/lab9/ROOMbasicsJava/app/src/main/java/pl/udu/uwr/pum/roombasicsjava/viewmodel/WordViewModel.java
package pl.udu.uwr.pum.roombasicsjava.viewmodel;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

import pl.udu.uwr.pum.roombasicsjava.model.Word;
import pl.udu.uwr.pum.roombasicsjava.model.WordRoomDatabase;

public class WordViewModel extends AndroidViewModel {

    private final WordRoomDatabase db;

    private final LiveData<List<Word>> allWords;

    public WordViewModel (Application application) {
        super(application);
        db = WordRoomDatabase.getDatabase(application);
        allWords = db.wordDao().getAlphabetizedWords();
    }

    public LiveData<List<Word>> getAllWords() { return allWords; }

    public void insert(Word word) {
        WordRoomDatabase.databaseWriteExecutor.execute(() -> db.wordDao().insert(word));
    }
}

File: 2022-2023/sourcecode/lab/lab9/ROOMbasicsKotlin/app/src/main/java/pl/udu/uwr/pum/roombasicskotlin/viewmodel/WordViewModel.kt
package pl.udu.uwr.pum.roombasicskotlin.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.launch
import pl.udu.uwr.pum.roombasicskotlin.model.Word
import pl.udu.uwr.pum.roombasicskotlin.model.WordRoomDatabase

class WordViewModel(application: Application) : AndroidViewModel(application) {
    private val db: WordRoomDatabase
    val getAllWords: LiveData<List<Word>>

    fun insert(word: Word) {
        viewModelScope.launch{
            db.wordDao().insert(word)
        }
    }

    init {
        db = WordRoomDatabase.getDatabase(application)
        getAllWords = db.wordDao().getAlphabetizedWords()
    }
}